src/models/local_search.rs

```diff
@@ -0,0 +1,405 @@
+//! Local autocompletion search
+//!
+
+use enclose::enclose;
+use futures::FutureExt;
+use http::Request;
+use num::{rational::Ratio, ToPrimitive};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DefaultOnError, NoneAsEmptyString};
+use url::Url;
+
+use localsearch::{self, LocalSearch as Searcher, DEFAULT_SCORE_THRESHOLD};
+
+use crate::{
+    constants::{CINEMETA_CATALOGS_URL, CINEMETA_FEED_CATALOG_ID},
+    models::{
+        common::{eq_update, Loadable},
+        ctx::Ctx,
+    },
+    runtime::{
+        msg::{Action, ActionLoad, ActionSearch, Internal, Msg},
+        Effect, EffectFuture, Effects, Env, EnvError, EnvFutureExt, UpdateWithCtx,
+    },
+};
+
+pub use imdb_rating::*;
+
+const INDEX_OPTIONS: IndexOptions = IndexOptions {
+    imdb_rating_weight: 0.5,
+    popularity_weight: 0.5,
+};
+
+/// The response returned when fetching the searchable items list.
+///
+/// Currently this is fetched from Cinemeta's `feed.json`
+#[derive(Deserialize, Serialize, Clone)]
+#[serde(transparent)]
+pub struct SearchableItemsResponse(pub Vec<Searchable>);
+
+#[derive(Copy, Clone)]
+pub struct IndexOptions {
+    imdb_rating_weight: f64,
+    popularity_weight: f64,
+}
+
+/// A searchable item
```

Comment:
Do such comments bring any value?

Reply:
yes, because we need to define what this struct does and document all fields.